modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/DeviceUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.common.util;
+
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.beans.CacheEntry;
+
+import javax.cache.Cache;
+import javax.cache.CacheConfiguration;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class for random token for XCEP and WSTEP
+ */
+public class DeviceUtil {
+
+    private static final String TOKEN_CACHE_MANAGER = "TOKEN_CACHE_MANAGER";
+    private static final String TOKEN_CACHE = "TOKEN_CACHE";
+    private static final long CACHE_DURATION = 15l;
+    private static boolean isContextCacheInitialized = false;
+
+    public static String generateRandomToken() {
+        return String.valueOf(UUID.randomUUID());
+    }
+
+    public static void persistChallengeToken(String token, String deviceID, String username) throws DeviceManagementException {
+
+        Object objCacheEntry = getCacheEntry(token);
+        CacheEntry cacheEntry;
+        if (objCacheEntry == null) {
+            cacheEntry = new CacheEntry();
+            cacheEntry.setUsername(username);
+        } else {
+            cacheEntry = (CacheEntry) objCacheEntry;
+        }
+        cacheEntry.setDeviceID(deviceID);
+        getTokenCache().put(token, cacheEntry);
+
+    }
+
```

Comment:
I do not see any place where an exception can occur, therefore what is the point of throwing a DeviceManagementException